source/extensions/common/wasm/context.cc

```diff
@@ -1638,6 +1638,9 @@ void Context::failStream(WasmStreamType stream_type) {
 WasmResult Context::sendLocalResponse(uint32_t response_code, std::string_view body_text,
                                       Pairs additional_headers, uint32_t grpc_status,
                                       std::string_view details) {
+  if (local_reply_hold_) {
```

Comment:
Needs tests

Reply:
Sorry for the delay, I've added a test that covers this line.